src/middlewared/middlewared/plugins/pool.py

```diff
@@ -696,9 +718,12 @@ def add(attr):
 
         dataset = await self.middleware.call('pool.dataset.query', [('id', '=', id)])
         if not dataset:
-            verrors.add('id', f'{id} does not exist', errno.ENOENT)
+            verrors.add('pool_dataset_update.id', f'{id} does not exist', errno.ENOENT)
         else:
             data['type'] = dataset[0]['type']
+            data['name'] = dataset[0]['name']
+            if data['type'] == 'VOLUME':
+                data['volsize'] = data['volsize'] if data.get('volsize') else dataset[0]['volsize']['rawvalue']
```

Comment:
Hmm, that will set the property again if its already set, I think we should avoid defining volsize